consensus/core/src/dag_state.rs

```diff
@@ -329,26 +382,33 @@ impl DagState {
         }
     }
 
-    // Write commits to store. Commits should be provided in commit order, meaning
-    // the last element in commits is the new last_commit.
-    pub(crate) fn write_commits(
-        &mut self,
-        commits: Vec<Commit>,
-        committed_blocks: Vec<VerifiedBlock>,
-    ) {
-        assert!(!commits.is_empty());
-        let last_commit = commits.last().unwrap().clone();
+    /// After each flush, DagState becomes persisted in storage and it expected to recover
+    /// all internal states from stroage after restarts.
+    pub(crate) fn flush(&mut self) {
```

Comment:
Let's have a unit test for the `flush` method as well

Reply:
optional: we could have some metrics to understand how much data get flushed and/or how much data retained in cache - just to confirm the expectations/behaviour of the nodes once deployed

Reply:
Added test and metrics.